tests/iter/tyieldintry.nim

```diff
@@ -1,9 +1,11 @@
 discard """
-targets: "c cpp"
+targets: "c"
 output: "ok"
 """
 var closureIterResult = newSeq[int]()
 
+# XXX Investigate why this fails now for 'nim cpp'
```

Comment:
I think that should be solved before merging this PR otherwise it introduces a new regression for non-arc code

Reply:
In general I agree but I'm overly busy and cooldome needs this patch.